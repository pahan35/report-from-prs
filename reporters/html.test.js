const MockDate = require('mockdate')

MockDate.set(new Date('2020-10-25T17:35:35'))
jest.mock('temp-write', () => jest.fn().mockImplementation(() => '/temp/path'))
jest.mock('open', () => jest.fn())

const {report} = require('./html')

const tempWrite = require('temp-write')
const open = require('open')

afterAll(() => {
  MockDate.reset()
})

test('html report', async () => {
  await report({
    sections: [
      [
        'Done',
        [
          {
            html_url: 'https://github.com/org/repo/pull/35',
            title: 'feat: feature 1',
          },
          {
            html_url: 'https://github.com/org/repo/pull/135',
            title: 'fix: critical bug',
          },
        ],
      ],
      [
        'WIP',
        [
          {
            html_url: 'https://github.com/org/repo/pull/5',
            title: 'feat: huge feature in progress',
          },
          {
            html_url: 'https://github.com/org/repo/pull/235',
            title: 'fix: low priority bug',
          },
        ],
      ],
    ],
    source: {forDays: 7},
  })
  expect(tempWrite).toHaveBeenCalledWith(expect.anything(), 'report.html')
  expect(tempWrite.mock.calls[0][0]).toMatchSnapshot()
  expect(open).toHaveBeenCalledWith('/temp/path')
})
