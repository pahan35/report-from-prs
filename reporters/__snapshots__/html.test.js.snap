// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`html report 1`] = `
"<!doctype html>
<html lang=\\"en\\">
  <head>
      <meta charset=\\"UTF-8\\">
      <meta name=\\"viewport\\"
             content=\\"width=device-width, user-scalable=no, initial-scale=1.0, maximum-scale=1.0, minimum-scale=1.0\\">
      <meta http-equiv=\\"X-UA-Compatible\\" content=\\"ie=edge\\">
      <title>Reports for  for repo  for 7 day(s)</title>
  </head>
  <body>
    <h1>
        Reports for
        <a href=\\"https://github.com/\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\"></a>
        for repo <a href=\\"https://github.com/\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\"></a>
        for 7 day(s) (since 10/18/2020, 5:35:35 PM till 10/25/2020, 5:35:35 PM)
    </h1>
    
      <h2>Done</h2>
      <ul>
          
            <li>feat: feature 1 [<a href=\\"https://github.com/org/repo/pull/35\\">pr<a>]</li>
          
            <li>fix: critical bug [<a href=\\"https://github.com/org/repo/pull/135\\">pr<a>]</li>
          
      </ul>
    
      <h2>WIP</h2>
      <ul>
          
            <li>feat: huge feature in progress [<a href=\\"https://github.com/org/repo/pull/5\\">pr<a>]</li>
          
            <li>fix: low priority bug [<a href=\\"https://github.com/org/repo/pull/235\\">pr<a>]</li>
          
      </ul>
    
  </body>
</html>
"
`;
